doc/developers-guide/Hooks-and-handlers.md

```diff
@@ -106,8 +95,19 @@ Rs = mongoose_acc:get(offline_messages, Acc1, []),
 
 ```
 
+### Sidenote: something deprecated
+
+Occassionally you may find in the MongooseIM source code some calls to `ejabberd_hooks:run/3`.
+Under the hood it calls the same handlers with an empty accumulator.
+This is deprecated and some day will be removed.
+
+### Error handling in hooks
+
+Hooks are meant to decouple modules; in other words, the caller signals that some event took place or that it intends to use a certain feature or a set of features, but how and if those features are implemented is beyond its interest.
+Hooks don't therefore use the "let it crash" approach. Instead it is rather like "fire-and-forget", more similar in principle to the `Pid ! signal` way.
```

Comment:
For that reason hooks don't use the "let it crash" approach. 
Instead it is rather like "fire-and-forget", more similar in principle to the `Pid ! signal` way.